File: app/src/main/java/com/example/tobias/werwolf_v1/SquareLayout.java
package com.example.tobias.werwolf_v1;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.LinearLayout;

public class SquareLayout extends LinearLayout {

    private final double mScale=1.0;

    public SquareLayout(Context context) {
        super(context);
    }

    public SquareLayout(Context context, AttributeSet attrs){
        super(context, attrs);
    }

    @Override
    protected void onMeasure(int widthMaesureSpec, int hightMeasureSpec){
        int width = MeasureSpec.getSize(widthMaesureSpec);
        int height = MeasureSpec.getSize(hightMeasureSpec);

        if (width > (int)((mScale * height) + 0.5)) {
            width = (int)((mScale * height) + 0.5);
        } else {
            height = (int)((width / mScale) + 0.5);
        }

        super.onMeasure(
                MeasureSpec.makeMeasureSpec(width, MeasureSpec.EXACTLY),
                MeasureSpec.makeMeasureSpec(height, MeasureSpec.EXACTLY)
        );
    }
}

File: app/src/main/java/com/example/tobias/werwolf_v1/SquareLayout.kt
package com.example.tobias.werwolf_v1

import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout

class SquareLayout : LinearLayout {
    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}

    override fun onMeasure(widthMaesureSpec: Int, hightMeasureSpec: Int) {
        var width = MeasureSpec.getSize(widthMaesureSpec)
        var height = MeasureSpec.getSize(hightMeasureSpec)
        if (width > (height + 0.5).toInt()) {
            width = (height + 0.5).toInt()
        } else {
            height = (width + 0.5).toInt()
        }
        super.onMeasure(
            MeasureSpec.makeMeasureSpec(width, MeasureSpec.EXACTLY),
            MeasureSpec.makeMeasureSpec(height, MeasureSpec.EXACTLY)
        )
    }
}